functions/src/controllers/locations.js

```diff
@@ -0,0 +1,146 @@
+            await location.delete();
+            if (!location) {
+                return res.status(NOT_FOUND).send({
+                    message: getStatusText(NOT_FOUND),
+                    status: 'error',
+                });
+            }
+            return res.status(OK).send({
+                data: null,
+                status: 'success',
+            });
+        } catch (error) {
+            return res.status(INTERNAL_SERVER_ERROR).send({
+                error,
+                message: getStatusText(INTERNAL_SERVER_ERROR),
+                status: 'error',
+            });
+        }
+    },
+
+    async getAll(req, res) {
+        try {
+            const query = await db.collection('locations');
+            const locations = [];
+            await query.get().then(querySnapshot => {
+                const docs = querySnapshot.docs;
+                for (const doc of docs) {
+                    const selectedItem = {
+                        id: doc.id,
+                        location: doc.data(),
+                    };
+                    locations.push(selectedItem);
+                }});
+            return res.status(OK).send({
+                data: { locations },
+                status: 'success',
+            });
+        } catch (error) {
+            return res.status(INTERNAL_SERVER_ERROR).send({
+                error,
+                message: getStatusText(INTERNAL_SERVER_ERROR),
+                status: 'error',
+            });
+        }
+    },
+
+    async getOne(req, res) {
+        try {
+            const document = db.collection('locations').doc(req.params.id);
+            if (!document) {
+                return res.status(NOT_FOUND).send({
+                    message: getStatusText(NOT_FOUND),
+                    status: 'error',
+                });
+            }
+            const documentData = await document.get();
+            const location = documentData.data();
+            return res.status(OK).send({
+                data: { location },
+                status: 'success',
+            });
+        } catch (error) {
+            return res.status(INTERNAL_SERVER_ERROR).send({
+                error,
+                message: getStatusText(INTERNAL_SERVER_ERROR),
+                status: 'error',
+            });
+        }
+    },
+
+    async updateLocation(req, res) {
+        try {
+            const document = db.collection('locations').doc(req.params.id);
+
+            if (!document) {
+                return res.status(NOT_FOUND).send({
+                    status: 'error',
+                    message: getStatusText(NOT_FOUND),
+                });
+            }
+            const location = await document.update(req.body);
+
+            return res.status(CREATED).send({
+                status: 'success',
+                data: { location, message: 'Comment successfully updated' },
+            });
```

Comment:
return { 
     status: 'success',
     message: ' Location successfully updated', 
     data: Location
}

Reply:
Ofcourse with the status code